packages/amplify-prompts/src/validators.ts

```diff
@@ -17,6 +17,11 @@ export const alphanumeric =
   (input: string) =>
     /^[a-zA-Z0-9]+$/.test(input) ? true : message;
 
+export const regexValidator =
```

Comment:
Bit: suggest naming this `matchRegex`. Also I don't think there should be a default regex. It should just be a required arg